hydra/extra/pytest_plugin.py

```diff
@@ -51,6 +51,11 @@ def _(
         sweep.config_name = config_name
         sweep.strict = strict
         sweep.overrides = overrides or []
+        sweep_dir_overrides = [r for r in sweep.overrides if "hydra.sweep.dir=" in r]
```

Comment:
I think it's best if you compose the config and get the run dir fro mhydra.sweep.dir in the resulting config.
this kind of logic is fragile and hacky.

Reply:
Thanks! Didn't think of that. will address this once we agree this is the best way to approach the issue.

Reply:
forgot to address this one, will fix this in a follow up commit

Reply:
roger.

Reply:
I didn't end using the ```compose``` API but used the utils from hydra internal to get the overrides value.

```compose``` API doesn't apply to the use case here since here I want to know if the tests has overriden ```sweep.dir```, if not, create a ```tmp_dir```